dhall/src/Dhall/Import.hs

```diff
@@ -510,20 +526,25 @@ loadImportWithSemanticCache
             let actualHash = Dhall.Crypto.sha256Hash bytesStrict
 
             if semanticHash == actualHash
-                then return ()
-                else do
-                    Status { _stack } <- State.get
-                    throwMissingImport (Imported _stack (HashMismatch {expectedHash = semanticHash, ..}))
+                then do
+                    let bytesLazy = Data.ByteString.Lazy.fromStrict bytesStrict
 
-            let bytesLazy = Data.ByteString.Lazy.fromStrict bytesStrict
+                    importSemantics <- case Dhall.Binary.decodeExpression bytesLazy of
+                        Left  err -> throwMissingImport (Imported _stack err)
+                        Right e   -> return e
 
-            importSemantics <- case Dhall.Binary.decodeExpression bytesLazy of
-                Left  err -> throwMissingImport (Imported _stack err)
-                Right e   -> return e
+                    return (ImportSemantics {..})
+                else do
+                    printWarning $
+                        makeHashMismatchMessage semanticHash actualHash
+                        <> "\n"
+                        <> "The interpreter has automatically fixed cached import\n"
```

Comment:
We can't really promise that, we can only try. Not sure how to put that into a message…